core/src/main/resources/error/error-classes.json

```diff
@@ -2109,11 +2103,6 @@
       "Unsupported component type <clz> in arrays."
     ]
   },
-  "_LEGACY_ERROR_TEMP_1104" : {
-    "message" : [
-      "The second argument should be a double literal."
```

Comment:
Are you saying this error is not suitable by re-using `INVALID_PARAMETER_VALUE`, we still need this error itself?

Reply:
In the fact, the expression must be a literal (foldable I think), and we can throw `DATATYPE_MISMATCH.NON_FOLDABLE_INPUT`. And then check that it is a double or a decimal (throw `DATATYPE_MISMATCH.UNEXPECTED_INPUT_TYPE`.

And one more thing, the entire checking logic should be moved to `checkInputDataTypes` instead of checking in a constructor.